kedro/config/common.py

```diff
@@ -130,6 +130,7 @@ def _load_config_file(config_file: Path, ac_template: bool = False) -> Dict[str,
         raise BadConfigException(f"Couldn't load config file: {config_file}") from exc
 
     except ParserError as exc:
+        assert exc.problem_mark is not None
```

Comment:
Starting from 0.600, mypy introduced strict optional typing--basically, if it's marked as `Optional`, it could be `None`. Since `ParserError.problem_mark` is typed as `Optional[Mark]`, we need to convince mypy that it indeed won't be `None` in this case, by adding an assert. This is standard practice, but if it gets too unwieldy (it's not here, just one assert statement), you can turn off optional typing as in https://mypy.readthedocs.io/en/stable/common_issues.html#unexpected-errors-about-none-and-or-optional-types.